src/charm.py

```diff
@@ -696,11 +717,15 @@ def _on_config_changed(self, _):
             try:
                 self._define_services()
                 self._define_ingresses()
-
+                ingress_ip = (
+                    f"Ingress IP(s): {', '.join(self._report_ingress_ips())}, "
+                    if self._report_ingress_ips()
+                    else ""
+                )
                 # It's not recommended to do this via ActiveStatus, but we don't
                 # have another way of reporting status yet.
-                msg = "Ingress with service IP(s): {}".format(
-                    ", ".join(self._report_service_ips())
+                msg = f"{ingress_ip}Service IP(s): {', '.join(self._report_service_ips())}".format(
+                    ingress_ip,
```

Comment:
This seems a little awkward as you've got a variable with a trailing space in it which means you have to concatenate without a space here (in case `ingress_ip` is empty. (Also, you're mixing f-strings and string formatting). Also, I think you're calling `self._report_ingress_ips()` twice in the block above? How about something like this:
```
msgs = []
ingress_ips = self._report_ingress_ips()
if ingress_ips:
    msgs.append(f"Ingress IP(s): {', '.join(ingress_ips)}
msgs.append(f"Service IP(s): {', '.join(self._report_service_ips())}")
msg = ", ".join(msgs)
```